internal/merkle/inmemory/merkle_tree.go

```diff
@@ -56,14 +54,10 @@ func (mt *MerkleTree) LeafCount() int64 {
 //
 // Returns the position of the leaf in the tree. Indexing starts at 1,
```

Comment:
Comment needs updating to match.

Reply:
I was a bit lazy with comments in the intermediate commits, as this file was deleted in the end. LMK if the final state looks good.